crates/runtime/src/logic.rs

```diff
@@ -80,6 +82,16 @@ impl<'a> VMLogic<'a> {
         }
         .build()
     }
+
+    pub fn get_executor_public_key(&mut self, register_id: u64) -> Result<()> {
+        self.registers
+            .set(self.limits, register_id, *&self.context.executor_public_key)
+    }
+
+    pub fn write_to_register(&mut self, register_id: RegisterId, data: &[u8]) -> Result<()> {
+        self.registers
+            .set(self.limits, register_id.as_usize() as u64, data)
+    }
```

Comment:
this doesn't seem to have any utility, as we can read from guest memory directly, and for the purposes of this patch, we shouldn't need this anyway

Reply:
That's fair - removed.